esp-hal-common/src/uart.rs

```diff
@@ -592,6 +592,65 @@ where
             .write(|w| unsafe { w.clkdiv().bits(divider).frag().bits(0) });
     }
 
+    #[cfg(esp32c6)]
+    fn change_baud(&self, baudrate: u32, clocks: &Clocks) {
+        // we force the clock source to be APB and don't use the decimal part of the
+        // divider
+        let clk = clocks.apb_clock.to_Hz();
+        let max_div = 0b1111_1111_1111 - 1;
+        let clk_div = ((clk) + (max_div * baudrate) - 1) / (max_div * baudrate);
+
+        // UART clocks are configured via PCR
+        let pcr = unsafe { &*esp32c6::PCR::PTR };
+
+        match self.uart.uart_number() {
```

Comment:
We have access the each Instances register block, so why do we need to match here?

Reply:
Because the register and field names are all different? I'm not sure I understand your question.

Reply:
Ah yeah missed that, oops. I think we can abstract this better, but its fine for now I think.